util/model_queries.go

```diff
@@ -98,6 +101,53 @@ func FindTaint(node *v1.Node, taintKey string) *v1.Taint {
 	return nil
 }
 
+// A pod may be assigned to multiple resource pools. The first one returned is considered the primary which will
```

Comment:
what is the use case covered by pods assigned to multiple resource pools?

Reply:
GPU instances. So a user can tell, please use "g4n.metal,g4dn.4xlarge" or some other sequences.
GPU resource pools are associated with specific GPU instance types (at least now).